smart-contracts/test/Unlock/behaviors/createLock.js

```diff
@@ -1,44 +1,30 @@
 const Units = require('ethereumjs-units')
-const Lock = artifacts.require('../Lock.sol')
-const Unlock = artifacts.require('../Unlock.sol')
+const Lock = artifacts.require('../../Lock.sol')
 
-let unlock
-
-contract('Lock', (accounts) => {
-  before(() => {
-    return Unlock.deployed()
-      .then(_unlock => {
-        unlock = _unlock
-        return unlock
-      })
-  })
-
-  describe('createLock', () => {
+exports.shouldCreateLock = function (accounts) {
+  describe('createLock', function () {
     let transaction
-    before(() => {
-      return unlock.createLock(
+    before(async function () {
+      transaction = await this.unlock.createLock(
         0, // keyReleaseMechanism: KeyReleaseMechanisms.Public
         60 * 60 * 24 * 30, // expirationDuration: 30 days
         Units.convert(1, 'eth', 'wei'), // keyPrice: in wei
         100 // maxNumberOfKeys
         , {
           from: accounts[0]
-        }).then((tx) => {
-        transaction = tx
-      })
+        })
     })
 
-    it('should have kept track of the Lock inside Unlock with the right balances', () => {
+    it('should have kept track of the Lock inside Unlock with the right balances', async function () {
       let lock = Lock.at(transaction.logs[0].args.newLockAddress)
       // This is a bit of a dumb test because when the lock is missing, the value are 0 anyway...
-      return unlock.locks(lock.address).then(([deployed, totalSales, yieldedDiscountTokens]) => {
-        assert(deployed)
-        assert.equal(totalSales.toNumber(), 0)
-        assert.equal(yieldedDiscountTokens.toNumber(), 0)
-      })
+      let [deployed, totalSales, yieldedDiscountTokens] = await this.unlock.locks(lock.address)
```

Comment:
+1 for using async/await. I should have done that earlier.... thanks!

Reply:
Good, I was a little hesitant changing styles like that without checking in with you first.  I do think this makes parts of the test much more readable.